webapp/channels/src/packages/mattermost-redux/src/actions/threads.ts

```diff
@@ -226,6 +226,16 @@ export function getThread(userId: string, teamId: string, threadId: string, exte
         } catch (error) {
             forceLogoutIfNecessary(error, dispatch, getState);
             dispatch(logError(error));
+
+            // Make sure the thread is marked as unfollowed as this API throws a 404 when the thread isn't followed
+            dispatch({
+                type: ThreadTypes.FOLLOW_CHANGED_THREAD,
+                data: {
+                    id: threadId,
+                    following: false,
+                },
+            });
+
```

Comment:
should we do this before `logError`? Just a thought nothing to change

Reply:
Not sure if it matters, 0/5 myself. Unless did we want to batch these actions?

Reply:
We cannot batch a pure action `FOLLOW_CHANGED_THREAD` with a thunk action `logError` so I guess this is fine.
Are they called pure actions or something I don't know :P

Reply:
I think this is a good change, albeit a just-in-case kind of thing, but does it need to be gated with a check for `error.status_code === 404` or something?